Add: Consistent Versioning of RPMs

tst-jblair opened the pull request:
Use the long version as the basis of the RPM filenames.

Test Procedure:
- Build and Stage-Install.
- Note the new file: /tmp/fusion_os_install/gee_long_version.txt
- Cd to earth_enterprise/rpms/
- Run: ./gradlew -PbuildOpenGee
- ls build/distributions/*rpm

Check and compare that filenames are as expected. You may edit /tmp/fusion_os_install/gee_long_version.txt and re-build with Gradle to check additional values.

Example expectations (the el6 part may change based on your OS):
5.2.1-2.release --> opengee-server-5.2.1.1.el6.rpm
5.2.2-0.alpha.DATESTAMP.GITHASH --> opengee-server-5.2.2.0.alpha.DATESTAMP.GITHASH-1.el6.rpm
5.2.2-1.beta --> opengee-server-5.2.2.1.beta-1.el6.rpm
5.2.2-2.beta.DATESTAMP.GITHASH --> opengee-server-5.2.2.2.beta.20171208.hashhash-1.el6.rpm
5.2.2-4.release --> opengee-server-5.2.2-1.el6.rpm

tst-ccamp commented on earth_enterprise/rpms/build.gradle:
```diff
@@ -44,7 +44,7 @@ apply plugin: "nebula.ospackage"
 
 def openGeeVersion =
-    new File(project.projectDir, '../src/fusion_version.txt').readLines().
+    new File('/tmp/fusion_os_install/gee_long_version.txt').readLines().
```
This needs to use the stagedInstallDir variable (defined below) to determine the path instead of hard-coding to /tmp/fusion_os_install

tst-ccamp commented on earth_enterprise/rpms/build.gradle:
```diff
@@ -55,7 +55,9 @@ def openGeeVersion =
         it // The first string that remains is the version string.
-    }
+    }.
+    replaceAll(/-.*release/, '').
```
Have you checked to make sure that stripping this out will still allow a "release" RPM to upgrade an alpha or beta RPM? The original reason we included "release" was because of the belief that without it RPM (cause it's dumb) will not consider it a higher version.

tst-ccamp commented on earth_enterprise/rpms/build.gradle:
```diff
@@ -55,7 +55,9 @@ def openGeeVersion =
-    }
+    }.
+    replaceAll(/-.*release/, '').
+    replaceAll('-', '.')
```
Looking at this a little more, I think for the sake of proper upgrades, we may have to truncate the "-2" or move it instead of just replacing the '-' with a '.'. Because it will consider 5.2.1.2 to be a newer version than 5.2.1 no matter what else comes after it.

tst-jblair replied:
I mentioned this to @tst-jcupp, but he decided that the larger priority was to match the RPM guidance in the Version Description document and get something working (given that there's some current process that uninstalls the existing version anyway?).  So, this PR is designed to meet the Version document as closely as possible (for RPMS - also ignoring .deb for now).

tst-ccamp commented on earth_enterprise/rpms/build.gradle:
```diff
@@ -55,18 +66,9 @@ def openGeeVersion =
-}
+    }.
+    replaceAll(/-.*release/, '').
+    replaceAll('-', '.')
```
Moving the discussion here, since the latest change ate the comments:
tst-ccamp an hour ago  Collaborator
Looking at this a little more, I think for the sake of proper upgrades, we may have to truncate the "-2" or move it instead of just replacing the '-' with a '.'. Because it will consider 5.2.1.2 to be a newer version than 5.2.1 no matter what else comes after it.
      @tst-jblair
tst-jblair an hour ago  Collaborator
I mentioned this to @tst-jcupp, but he decided that the larger priority was to match the RPM guidance in the Version Description document and get something working (given that there's some current process that uninstalls the existing version anyway?). So, this PR is designed to meet the Version document as closely as possible (for RPMS - also ignoring .deb for now).

tst-jblair commented:
Switched to using the variable as requested.

Removed a merge conflict that appeared with some other changes within the last day.

There is still the issue of "release" rpms not being successors of their prior "alpha"/"beta" rpms. My interpretation of our discussions is that this is not great, but also not a blocker. Especially as this version of the code matches (as closely as possible) the descriptions given in the Versioning guide. A follow-on task to update the guide, and then these rpms is recommended.

tst-ccamp commented:
@tst-jblair In addition to recommending, actually writing the follow-on task would be a good solution :)

tst-jblair commented:
Done: https://github.com/google/earthenterprise/issues/650

tst-ccamp commented:
Tested the changes and reviewed updates. LGTM
